controllers/awsmachine_controller.go

```diff
@@ -912,3 +918,36 @@ func (r *AWSMachineReconciler) indexAWSMachineByInstanceID(o client.Object) []st
 
 	return nil
 }
+
+func (r *AWSMachineReconciler) ensureStorageTags(ec2svc services.EC2MachineInterface, instance *infrav1.Instance, machine *infrav1.AWSMachine) {
+	annotations := make(map[string]interface{})
+	for _, volumeID := range instance.VolumeIDs {
+		if _, ok := annotations[volumeID]; ok {
```

Comment:
`annotations` is an empty map, this will always be false.

`machineAnnotation` and `additionalTags` needs to be compared here. See `ensureTags()` as an example.

Reply:
If i do that check outside if condition I am getting below error 
```
[manager] panic: interface conversion: interface {} is map[string]interface {}, not string
[manager] 
[manager] goroutine 673 [running]:
[manager] sigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSMachineReconciler).tagsChanged(0xc000200230, 0xc000b3f7d0, 0xc000cb3830, 0x40e278, 0x10, 0x212cd40, 0x1)
[manager] 	/Users/swamyan/go/src/sigs.k8s.io/cluster-api-provider-aws/controllers/awsmachine_tags.go:111 +0x5c8
[manager] sigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSMachineReconciler).ensureVolumeTags(0xc000200230, 0x2aa0490, 0xc000992630, 0xc0006291e0, 0xc000b3f7d0, 0xc000cb3830, 0x0, 0x1, 0x100000000000002)
[manager] 	/Users/swamyan/go/src/sigs.k8s.io/cluster-api-provider-aws/controllers/awsmachine_tags.go:76 +0x45
[manager] sigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSMachineReconciler).ensureStorageTags(0xc000200230, 0x2aa0490, 0xc000992630, 0xc0003ce3c0, 0xc00082e840)
[manager] 	/Users/swamyan/go/src/sigs.k8s.io/cluster-api-provider-aws/controllers/awsmachine_controller.go:932 +0x225
[manager] sigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSMachineReconciler).reconcileNormal(0xc000200230, 0x2a7e838, 0xc000cb37a0, 0xc0000bbf40, 0x2aa33c8, 0xc00068cd20, 0x2aa7278, 0xc00068cd20, 0x2aa4610, 0xc00068cd20, ...)
[manager] 	/Users/swamyan/go/src/sigs.k8s.io/cluster-api-provider-aws/controllers/awsmachine_controller.go:589 +0xe25
[manager] sigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSMachineReconciler).Reconcile(0xc000200230, 0x2a7e838, 0xc000cb37a0, 0xc000b996a9, 0x7, 0xc0004f3ec0, 0x19, 0xc000cb3700, 0x0, 0x0, ...)
[manager] 	/Users/swamyan/go/src/sigs.k8s.io/cluster-api-provider-aws/controllers/awsmachine_controller.go:201 +0x8b4
[manager] sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0007be280, 0x2a7e790, 0xc000b311c0, 0x239b1c0, 0xc0000e0e00)
[manager] 	/Users/swamyan/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:298 +0x30d
[manager] sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0007be280, 0x2a7e790, 0xc000b311c0, 0x5d4a00)
[manager] 	/Users/swamyan/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:253 +0x205
[manager] sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc000690070, 0xc0007be280, 0x2a7e790, 0xc000b311c0)
[manager] 	/Users/swamyan/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:214 +0x6b
[manager] created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
[manager] 	/Users/swamyan/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0/pkg/internal/controller/controller.go:210 +0x425
[manager] Exiting with code 2
```

Reply:
jsonAnnotation in `machineAnnotationJSON` func is giving the value as null in map , "{\"vol-0e5e5b65521f6e259\":{\"foo\":\"bar\",\"rootDeviceID\":\"volume-id\",\"rootDeviceSize\":\"40\"}}"=null